WordPress/gutenberg
# Data views: Align page headers

## What?
Visually and structurally align data views page headers.

## Why?
There are currently several disparities between the Patterns page header and other page headers in the Site Editor. This PR updates both `Header` and `PatternsHeader` components to bring them into alignment.

* Title + Actions live in the same container with optional description below.
* Header font size/weight matches (I could go either way on which size).
* Description styling matches.
* Action buttons use new 40px size rather than non-standard 36px.
* Padding matches.

### Before

<img width="1199" alt="Screenshot 2024-05-29 at 18 33 18" src="https://github.com/WordPress/gutenberg/assets/846565/9fdd0b34-ccba-478b-8c90-c87e6cdd12b7">
<img width="1203" alt="Screenshot 2024-05-29 at 18 33 26" src="https://github.com/WordPress/gutenberg/assets/846565/67ab78b8-994f-42e7-a443-6bc7dc4d0426">
<img width="1201" alt="Screenshot 2024-05-29 at 18 33 35" src="https://github.com/WordPress/gutenberg/assets/846565/3297074e-3182-4b30-818f-299a79a33584">


### After

<img width="1201" alt="Screenshot 2024-05-29 at 18 31 50" src="https://github.com/WordPress/gutenberg/assets/846565/b3d0506f-1559-4359-89c7-2a7de8201027">
<img width="1202" alt="Screenshot 2024-05-29 at 18 31 59" src="https://github.com/WordPress/gutenberg/assets/846565/1d799adf-b012-4c84-9ef2-a59e95483ea1">
<img width="1200" alt="Screenshot 2024-05-29 at 18 32 09" src="https://github.com/WordPress/gutenberg/assets/846565/a4d2b792-648f-4e86-83e1-f65d0cde2991">

### Testing instructions
* Navigate to Pages, Patterns, and Templates pages in the Site editor.
* Ensure the headers match visually and structurally across.
